python/lsst/ap/association/dia_collection.py

```diff
@@ -188,15 +232,51 @@ def match(self, dia_source_catalog, scores):
         dia_source_catalog : an lsst.afw.SourceCatalog
             A contiguous catalog of dia_sources for which the set of scores
             has been computed on with DIAObjectCollection.score.
-        scores : (N DIAObject by M DIASource) float array
-            A ndarray containing the scores between each DIAObject and
-            each DIASource.
+        score_struct : lsst.pipe.base.Struct
+            A struct containing two arrays: scores DIAObjects, indices of]
+            DIAObject mached to. Both arrays should be the same length as the
+            source catalog.
 
         Returns
         -------
         Indices of newly updated and created DIAObjects
         """
-        pass
+
+        used_dia_object = np.zeros(len(self.dia_objects), dtype=np.bool)
+        used_dia_source = np.zeros(len(dia_source_catalog), dtype=np.bool)
+
+        n_previous_objects = len(self.dia_objects)
+
+        updated_and_new_dia_objects = []
+
+        score_args = score_struct.scores.argsort(axis=None)
+        for score_idx in score_args:
+            if not np.isfinite(score_struct.scores[score_idx]):
+                break
```

Comment:
It should be break. Because of the sort all of the values past the first NaN are also NaN so we don't need to loop over them. I have added a comment to reflect this.